core/src/test/resources/sql/sub-query.iq

```diff
@@ -1174,7 +1173,7 @@ EnumerableCalc(expr#0..3=[{inputs}], expr#4=[IS NULL($t3)], expr#5=[null:BOOLEAN
     EnumerableLimit(fetch=[1])
       EnumerableSort(sort0=[$0], dir0=[DESC])
         EnumerableAggregate(group=[{0}], c=[COUNT()])
-          EnumerableCalc(expr#0..2=[{inputs}], expr#3=[true], cs=[$t3])
+          EnumerableCalc(expr#0..2=[{inputs}], expr#3=[CAST($t0):INTEGER], expr#4=[IS NOT NULL($t3)], cs=[$t4])
             EnumerableTableScan(table=[[scott, DEPT]])
 !plan
```

Comment:
Are these plan changes desirable?

Reply:
Has been fixed in https://issues.apache.org/jira/browse/CALCITE-5769. I have rolled back this change.